scripts/provision_credentials.py

```diff
@@ -0,0 +1,169 @@
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info('AWS secret %s doesn\'t exist. No changes will be made.', secret_name)
+            else:
+                raise
+        else:
+            assert response['Name'] == secret_name
+            logger.info('Successfully deleted AWS secret %s.', secret_name)
+
+    def _destroy_service_account_creds(self, service_account_email):
+        try:
+            creds = self.secrets_manager.get_secret_value(
+                SecretId=config.secrets_manager_secret_name('indexer', 'google_service_account')
+            )
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                logger.info('Secret already deleted, cannot get key_id for %s', service_account_email)
+                return
+            else:
+                raise
+        else:
+            key_id = json.loads(creds['SecretString'])['private_key_id']
+            service = get_google_service()
+            try:
+                service.projects().serviceAccounts().keys().delete(
+                    name='projects/-/serviceAccounts/' + service_account_email + '/keys/' + key_id).execute()
+            except HttpError as e:
+                if e.resp.reason != 'Not Found':
+                    raise
+            logger.info("Successfully deleted service account key with id '%s' for user '%s'",
+                        key_id, service_account_email)
+
+
+if __name__ == "__main__":
+    credentials_provisioner = CredentialsProvisioner()
+    provision_parser = argparse.ArgumentParser(add_help=False)
+    group = provision_parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--build', '-b', action='store_true', dest='build',
+                       help='Create credentials instead of destroying them. This action is idempotent.')
+    group.add_argument('--destroy', '-d', action='store_false', dest='build',
+                       help='Destroy credentials instead of building them. This action is idempotent.')
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(help='Specify action', dest='action')
+    subparsers.required = True
+    email_parser = subparsers.add_parser('google-key', parents=[provision_parser],
+                                         help='Create Google service account key and store in an AWS secret.')
+    email_parser.set_defaults(func=credentials_provisioner.provision_google_from_args)
+    email_parser.add_argument('email', type=str,
+                              help='Email address for the Google service account '
+                                   'for which to provision credentials')
+    subparsers.add_parser('hmac-key', parents=[provision_parser],
+                          help='Create a random HMAC key and store in an AWS secret.'
+                          ).set_defaults(func=credentials_provisioner.provision_hmac_from_args)
```

Comment:
```suggestion
                          ).set_defaults(func=CredentialsProvisioner.provision_hmac_from_args)
```

Reply:
Not a  fan of the chaining btw because it is asymmetric with the other subparser.

Reply:
Ok I removed it for symmetry